dolphin/api/common_utils.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from oslo_log import log
+
+LOG = log.getLogger(__name__)
+
+
+def remove_invalid_options(context, search_options, allowed_search_options):
```

Comment:
How about combining line 21 to 27 into:
for opt in set(search_options) - set(allowed_search_options):
        del search_options[opt]
its a suggestion, you can decide. Both code will work

Reply:
It would work, but want to capture and log unknown_options ..

Reply:
suggest rename the file to api_uitls, because common folder is under api folder.